sklearn/utils/estimator_checks.py

```diff
@@ -1521,22 +1521,35 @@ def check_estimators_overwrite_params(name, estimator_orig):
 
 
 @ignore_warnings(category=(DeprecationWarning, FutureWarning))
-def check_no_fit_attributes_set_in_init(name, Estimator):
-    """Check that Estimator.__init__ doesn't set trailing-_ attributes."""
-    # this check works on classes, not instances
-    estimator = Estimator()
-    for attr in dir(estimator):
-        if attr.endswith("_") and not attr.startswith("__"):
-            # This check is for properties, they can be listed in dir
-            # while at the same time have hasattr return False as long
-            # as the property getter raises an AttributeError
-            assert_false(
-                hasattr(estimator, attr),
-                "By convention, attributes ending with '_' are "
-                'estimated from data in scikit-learn. Consequently they '
-                'should not be initialized in the constructor of an '
-                'estimator but in the fit method. Attribute {!r} '
-                'was found in estimator {}'.format(attr, name))
+def check_no_attributes_set_in_init(name, estimator):
+    """Check that parameters are correctly setted during init and
+       there is no additional attribute apart from init parameters. .
+    """
+
+    if (name in ["GaussianProcess", "RandomizedLasso",
+                 "RandomizedLogisticRegression",
+                 "RandomizedPCA"]):
+        # To not check decorated (eg: deprecated) estimator
```

Comment:
I don't think this is quite right. You're not testing for decoration (nor am I sure it is possible to do so).

Reply:
Btw, the `deprecated_original` attribute is present, but on `GaussianProcess.__init__`, `RandomizedLasso.__init__`, etc.